include/highfive/H5DataType.hpp

```diff
@@ -80,7 +92,58 @@ class AtomicType : public DataType {
 
     typedef T basic_type;
 };
-}
+
+
+///
+/// \brief A structure representing a set of fixed-length strings
+///
+/// Although fixed-len arrays can be created 'raw' without the need for
+/// this structure, to retrieve results efficiently it must be used.
+///
+template <std::size_t N>
+class FixedLenStringArray {
+  public:
+    FixedLenStringArray() = default;
+
+    ///
+    /// \brief Create a FixedStringArray from a raw contiguous buffer
+    ///
+    FixedLenStringArray(const char array[][N], std::size_t length);
+
+    ///
+    /// \brief Create a FixedStringArray from a sequence of strings.
+    ///
+    /// Such conversion involves a copy, original vector is not modified
+    ///
+    FixedLenStringArray(const std::string* iter_begin, const std::string* iter_end);
+
+    explicit FixedLenStringArray(const std::vector<std::string> & vec);
+
+    FixedLenStringArray(const std::initializer_list<std::string> &);
+
+    ///
+    /// \brief Append an std::string to buffer structure
+    ///
+    void push_back(const std::string&);
+
+    ///
+    /// \brief Retrieve a string from the structure as std::string
+    ///
+    std::string operator[](std::size_t index) const;
+
+
+    // Container API
+    inline std::size_t size() const noexcept { return datavec.size(); }
+    inline char* data() const { return const_cast<char*>(datavec[0].data()); }
+    void resize(std::size_t new_size) { datavec.resize(new_size); }
```

Comment:
In case you want this class to mimic std::vector, there is also `void std::vector::resize(size_type n, const value_type& val)`. But we shouldn't add it if unused/tested in the project.

Reply:
I dont think that's required API for this class.